engine/src/flutter/lib/web_ui/lib/src/engine/canvaskit/painting.dart

```diff
@@ -323,21 +323,36 @@ class CkFragmentProgram implements ui.FragmentProgram {
 
   @override
   ui.FragmentShader fragmentShader() {
-    return CkFragmentShader(name, effect, floatCount, textureCount);
+    return CkFragmentShader(name, effect, this);
+  }
+
+  int _getShaderIndex(String name, int index) {
+    int result = 0;
+    for (final uniform in uniforms) {
+      if (uniform.name == name) {
+        if (index < 0 || index >= uniform.floatCount) {
+          throw IndexError.withLength(index, uniform.floatCount);
+        }
+        break;
```

Comment:
Correct me if I'm wrong, but it seems like we might need to do `result += index;` here before we break, right? Or maybe just `return result + index;` rather than breaking? The index is the offset into the uniform, and at this point `result` is pointing at the beginning of the uniform, so we need to point it at the position inside the uniform?